d/d.bzl

```diff
@@ -47,6 +47,14 @@ def _create_setup_cmd(lib, deps_dir):
       deps_dir + "/" + lib.basename + "\n"
   )
 
+def _files_directory(files):
+  """Returns the shortest parent directory of a list of files."""
+  dir = files[0].dirname
+  for f in files:
+    if len(dir) > len(f.dirname):
```

Comment:
Does it matter that this might not be the parent directory of all the files?  E.g., `["y/foo", "z/bar"]` will return "y".